sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/ReadBlobArgumentBinding.cs

```diff
@@ -39,5 +75,28 @@ public static async Task<Stream> TryBindStreamAsync(BlobBaseClient blob, Cancell
 
             return rawStream;
         }
+
+        private static async Task<FunctionDataCacheKey> GetFunctionDataCacheKey(BlobWithContainer<BlobBaseClient> blob, CancellationToken cancellationToken)
+        {
+            // To be strongly consistent, first check the latest version present in blob storage;
+            // query for that particular version in the cache.
+            BlobProperties properties = await blob.BlobClient.FetchPropertiesOrNullIfNotExistAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
keep in mind that this call and OpenRead might see different state of the blob. These are two separate calls to storage backend, so there might be a race condition here.
It might be better to fetch properties first and openRead with retrieved Etag later, so that both are assured to see same view of the blob (or fail).